src/SFA.DAS.ApprenticeCommitments/Data/Models/Registration.cs

```diff
@@ -29,6 +30,17 @@ private Registration()
             CommitmentsApprovedOn = commitmentsApprovedOn;
             Email = email;
             Apprenticeship = apprenticeship;
+
+            // The commitment statement isn't technically added until the apprentice
+            // confirms their identity, however it's possible that the apprentice
+            // never does.  This pretence will trigger events that Approvals can use
+            // to prompt the apprentice when the confirmation is overdue.
+            AddDomainEvent(
+                new CommitmentStatementAdded(
+                    new CommitmentStatement(
+                        commitmentsApprenticeshipId,
+                        commitmentsApprovedOn,
+                        Apprenticeship)));
```

Comment:
Shouldn't this be the DomainEvent RegistrationAdded, it will culminate in the publishing of the Event ConfirmationCommencedEvent, but from the domain POV it's a new registration and although we don't differentiate these externally, they are different events and may end up publishing different external messages in the future.

When we verify the apprentice for the first time (we create a CommitmentStatement) therefore shouldn't we be publishing the domain Event CommitmentStatementAdded in that flow (as well as when we renew). This keeps our domain model consistent

Reply:
Yes, I like this.  Even if it ends up publishing the same event it is triggered by a different domain process and should be modelled as such.  I'll rework it.